egs/wsj/s5/steps/libs/nnet3/train/frame_level_objf/common.py

```diff
@@ -489,27 +562,38 @@ def combine_models(dir, num_iters, models_to_combine, egs_dir,
     context_opts = "--left-context={lc} --right-context={rc}".format(
         lc=left_context, rc=right_context)
 
+    if use_multitask_egs:
+        combine_egs = "scp:{0}/combine.egs.1.scp".format(egs_dir)
+    else:
+        combine_egs = "ark:{0}/combine.egs".format(egs_dir)
+
+    multitask_egs_opts = common_train_lib.get_multitask_egs_opts(
+                             egs_dir,
+                             egs_prefix="combine.",
+                             use_multitask_egs=use_multitask_egs)
     common_lib.run_job(
         """{command} {combine_queue_opt} {dir}/log/combine.log \
                 nnet3-combine --num-iters=80 \
                 --enforce-sum-to-one={hard_enforce} \
```

Comment:
I think it is better to have separate combined model per output and do lbfgs optimization per task.
It is possible that the objective value for one of the outputs would be dominant and it doesn't work well for others.
What is your idea?